float32 action
float32 score
float32 reward